src/common/arrstr.cpp

```diff
@@ -26,6 +26,12 @@
 #include <algorithm>
 #include <functional>
 #include "wx/afterstd.h"
+#include "wx/regex.h"
+
+#ifdef __WXMSW__
```

Comment:
Perhaps` __WINDOWS__` should be used here instead of `__WXMSW__`, as this is not GUI related? TBH, I always confuse the two defines, all I know that the difference somehow affects the MS Windows GTK backend...

OTOH, I am quite sure that `#include "shlwapi.h`" should be changed to `#include <shlwapi.h>`.

Reply:
Mhm, looking at the failed tests, it seems that the issue with ` __WINDOWS__` vs  `__WXMSW__` is not that simple, as (some of) the tests do not define `__WINDOWS__`?

I based my assumption on [this](http://docs.wxwidgets.org/3.1/page_cppconst.html)
>`__WXMSW__` | GUI using Windows Controls.  Notice that for compatibility reasons, this symbol is defined for  console applications under Windows as well, but it should only be used  in the GUI code while` __WINDOWS__` should be used for the platform tests.

>`__WINDOWS__` | Any Windows platform, using any port (see also `__WXMSW__`)

Perhaps someone more knowledgeable than me will see this and explain which define should be used here...